common/src/main/java/com/wynntils/models/aspects/AspectModel.java

```diff
@@ -27,10 +52,120 @@ public void registerDownloads(DownloadRegistry registry) {
         aspectInfoRegistry.registerDownloads(registry);
     }
 
+    @SubscribeEvent
+    public void onContentSet(ContainerSetContentEvent.Pre event) {
+        Container currentContainer = Models.Container.getCurrentContainer();
+
+        List<Integer> equippedSlots = new ArrayList<>();
+        List<Integer> ownedSlots;
+
+        if (currentContainer instanceof AspectsContainer aspectsContainer) {
+            equippedSlots = new ArrayList<>(aspectsContainer.getEquippedSlots());
+            ownedSlots = new ArrayList<>(aspectsContainer.getAspectBounds().getSlots());
+        } else if (currentContainer instanceof RaidRewardChestContainer rewardChestContainer) {
+            ownedSlots = new ArrayList<>(rewardChestContainer.getAspectBounds().getSlots());
+        } else if (currentContainer instanceof RaidRewardContainer rewardContainer) {
+            ownedSlots = new ArrayList<>(rewardContainer.getRewardBounds().getSlots());
+        } else {
+            return;
+        }
```

Comment:
'getOwnedSlotsFromContainer` and null check the return value. Much more readable.